coordinator/src/main/scala/filodb/coordinator/v2/FiloDbClusterDiscovery.scala

```diff
@@ -0,0 +1,136 @@
+package filodb.coordinator.v2
+
+import java.net.InetAddress
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.pattern.ask
+import akka.util.Timeout
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.{CancelableFuture, Scheduler}
+import monix.reactive.Observable
+import net.ceedubs.ficus.Ficus._
+
+import filodb.coordinator.{CurrentShardSnapshot, FilodbSettings, ShardMapper}
+import filodb.core.DatasetRef
+
+class FiloDbClusterDiscovery(settings: FilodbSettings, system: ActorSystem,
```

Comment:
Can we reuse the logic from ``K8sStatefulSetShardAssignmentStrategy`` by extending that interface possibly? If no, we need to take the corresponding tests from from ``ShardAssignmentStrategySpec`` so any changes to this logic gets unit tested.

Reply:
Ah - makes sense. I copied the logic since I thought we'd be deleting the `ShardAssignmentStrategy` when we move over and everything stabilizes. Can look into unit tests.

Reply:
Created new FiloDbClusterDiscoverySpec and added unit tests there for the shard assignment